jaeger-core/src/main/java/io/jaegertracing/internal/samplers/HttpSamplingManager.java

```diff
@@ -28,20 +28,23 @@
 
 @ToString
 public class HttpSamplingManager implements SamplingManager {
-  public static final String DEFAULT_HOST_PORT = "localhost:5778";
-  private final String hostPort;
+  public static final String DEFAULT_SERVER_HOST = "localhost";
+  public static final int DEFAULT_SERVER_PORT = 5778;
+  public static final String DEFAULT_SERVER_URL =
+          "http://" + DEFAULT_SERVER_HOST + ":" + DEFAULT_SERVER_PORT + "/sampling";
+  private final String serverUrl;
 
   @ToString.Exclude private final Gson gson = new Gson();
 
   /**
-   * This constructor expects running sampling manager on {@link #DEFAULT_HOST_PORT}.
+   * This constructor expects running sampling manager on {@link #DEFAULT_SERVER_URL}.
    */
   public HttpSamplingManager() {
-    this(DEFAULT_HOST_PORT);
+    this(DEFAULT_SERVER_URL);
   }
 
-  public HttpSamplingManager(String hostPort) {
-    this.hostPort = hostPort != null ? hostPort : DEFAULT_HOST_PORT;
+  public HttpSamplingManager(String serverUrl) {
```

Comment:
It seems like a breaking change. 

We can keep `hostport`. `HttpBaggageRestrictionManagerProxy` also accepts `hostport`. Only HTTP sender accepts full URL e.g. `http://jaeger-collector:111/api/traces`. 

Host port is easier to configure. Not sure if there are any advantages of using full URL.

Reply:
Agreed - the primary driver of using ENDPOINT was to keep this in-line with the Go library.  If that's not a concern, then this is something I can change.